lib/reline/line_editor.rb

```diff
@@ -235,7 +235,6 @@ def reset_variables(prompt = '', encoding:)
     @vi_waiting_operator_arg = nil
     @completion_journey_state = nil
     @completion_state = CompletionState::NORMAL
-    @completion_occurs = false
```

Comment:
Why is this instance variable initialization no longer necessary?

Reply:
The actual scope of `@completion_occurs` is very narrow. Removing this initialization will represent the narrow scope clearer.
```ruby
def input_key(key)
  @completion_occurs = false # initialize here
  calls process_key() or normal_key() # completion might occur
  unless @completion_occurs; end # use it here

  # after here, @completion_occurs is no longer necessary.
  # maybe we can remove instance variable, but I think it's too much
end
```

Reply:
I see. Thanks!